### DISCUSSION TITLE
Smartsubscriptions and prisma relations

### DISCUSSION THREAD

**Discussion by aumer-amr** [hayes/pothos#1155] [Q&A]
I'm having difficulties understanding how smartsubscriptions work with prisma relations.
Let's say I have an object called Inventory and an object InventoryItem.

```typescript
schemaBuilder.prismaObject("Inventory", {
	fields: (t) => ({
		id: t.exposeID("id"),
		updatedAt: t.expose("updatedAt", {
			type: "DateTime",
		}),
		items: t.relation("items"),
	})
});
```

```typescript
schemaBuilder.prismaObject("InventoryItem", {
	fields: (t) => ({
		id: t.exposeID("id"),
		inventory: t.relation("inventory"),
		name: t.exposeString("name")
		createdAt: t.expose("createdAt", {
			type: "DateTime",
		})
	})
});
```

For this I build a query for Inventory that has a smart subscription, and start listening to that. However when I create a new item using a mutation that creates a new InventoryItem, how do I trigger the update on the inventory subscription?

**Comment by hayes** [comment]
The smart subscription plugin doesn't really know or care about relations.   It's basically just a way to say when x event is triggered, re-exute the resolvers for this field.

If you want to re-load a list when an item is created, you just need to to subscribe to an event that is triggered when items are created on your relation field

**Reply by hayes** [reply]
without a lot more context on how you are creating events its hard to give a more detailed answer
